db/version_set.cc

```diff
@@ -5329,6 +5402,28 @@ Status VersionSet::WriteCurrentStateToManifest(
     }
   }
 
+  {
+    // Save WALs.
+    if (!wals_.load()->GetWals().empty()) {
+      std::unique_ptr<VersionEdit> wal_additions(new VersionEdit());
+      for (const auto& wal : wals_.load()->GetWals()) {
```

Comment:
Not sure if I follow the intent of using atomic variable here.
`wals_` is `std::atomic<WalSet*>`. It means the pointer is atomic, but not the object pointed to by the pointer.
This piece of code is outside db mutex, and the `WalSet` object needs to be protected. What we do with other VersionStorage information is to copy them.

Reply:
Will your future changes modify `wals_` dynamically, i.e. let the pointer point to different `WalSet` objects? If so, how do you reason about the correctness here?

Reply:
I thought about the synchronization again, I should leverage release-acquire ordering of the atomic operation (as described in `Release-Acquire ordering` section in https://en.cppreference.com/w/cpp/atomic/memory_order). 

No, only the `ProcessManifestWrite` will update `wals_` after recovery.

Reply:
Actually, there is no need to use atomic operation to establish the happens-before relationship between the recover thread and the LogAndApply thread, I updated the comment of wals_ to explain the reason behind this.